src/libasr/rng.cpp

```diff
@@ -0,0 +1,14 @@
+#ifndef LFORTRAN_INTRINSICS_H
+#include <libasr/runtime/lfortran_intrinsics.h>
+#endif
+#include <random>
+
+extern "C" {
+    LFORTRAN_API double _lfortran_random() {
+        static std::random_device rd;
+        static std::mt19937 generator(rd());
+        static std::uniform_real_distribution<double> distribution(0.0, 1.0);
+
+        return distribution(generator);
+    };
```

Comment:
This must be implemented in the runtime library. However we use C there, not C++, so must be implemented in C.

Reply:
In `libasr/runtime` or `runtime` ?